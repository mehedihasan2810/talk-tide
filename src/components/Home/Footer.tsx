import Link from 'next/link'
import React from 'react'

const Footer = () => {
  return (
    <footer className="fixed bottom-0 left-0 flex w-full flex-wrap items-center justify-center gap-4 border-t border-t-primary/20 p-4 text-primary/50 sm:gap-6">
    <Link className="hover:text-primary" href="#">
      Privacy Policy
    </Link>
    <Link className="hover:text-primary" href="#">
      Terms
    </Link>
    <Link className="hover:text-primary" href="#">
      Cookies Policy
    </Link>
    <Link className="hover:text-primary" href="#">
      {" "}
      Â© Talk Tide 2023
    </Link>
  </footer>
  )
}

export default Footer