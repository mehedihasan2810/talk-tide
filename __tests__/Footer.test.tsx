import Footer from "@/components/Home/Footer";
import { render, screen } from "@testing-library/react";

describe("Footer Component", () => {
  test("Should render properly", () => {
    // ARRANGE
    const { asFragment } = render(<Footer />);

    // ACT
    // Iterate through the expected link names and ensure they are present in the rendered component
    ["Privacy Policy", "Terms", "Cookies Policy", "Â© Talk Tide 2023"].forEach(
      (name) => {
        expect(screen.getByRole("link", { name })).toBeInTheDocument();
      },
    );

    // ASSERT
    // Take a snapshot to compare against future changes
    expect(asFragment()).toMatchSnapshot();
  });
});
