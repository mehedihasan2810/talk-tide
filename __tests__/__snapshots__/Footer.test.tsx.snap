// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component Should render properly 1`] = `
<DocumentFragment>
  <footer
    class="fixed bottom-0 left-0 flex w-full flex-wrap items-center justify-center gap-4 border-t border-t-primary/20 p-2 text-primary/50 sm:gap-6 sm:p-4"
  >
    <a
      class="hover:text-primary"
      href="#"
    >
      Privacy Policy
    </a>
    <a
      class="hover:text-primary"
      href="#"
    >
      Terms
    </a>
    <a
      class="hidden hover:text-primary sm:inline-block"
      href="#"
    >
      Cookies Policy
    </a>
    <a
      class="hover:text-primary"
      href="#"
    >
       Â© Talk Tide 2023
    </a>
  </footer>
</DocumentFragment>
`;
